go/arrow/csv/common.go

```diff
@@ -104,13 +104,26 @@ func WithCRLF(useCRLF bool) Option {
 	}
 }
 
-func WithHeader() Option {
+func WithHeader(useHeader bool) Option {
 	return func(cfg config) {
 		switch cfg := cfg.(type) {
 		case *Reader:
-			cfg.header = true
+			cfg.header = useHeader
 		case *Writer:
-			cfg.header = true
+			cfg.header = useHeader
+		default:
+			panic(fmt.Errorf("arrow/csv: unknown config type %T", cfg))
+		}
+	}
+}
+
+func WithNullString(null string) Option {
```

Comment:
we should probably document (here or for both `Reader` and `Writer`) what is the default Null string and what happens when such a string is encountered.